02-getting-account-data/index.js

```diff
@@ -0,0 +1,38 @@
+import { SDK } from 'NEWSDKNAME'
+
+// SDK exposes all you need like anchor-link does, users don't need to import from core directly, if they need something they do
+// import { Account } from 'NEWSDKNAME'
+
+const sdk = new SDK({ url: 'https://eos.greymass.com' }) // loaded with default plugins (something like if (browser) { plugins = [anchor-browser, atomichub] })
+
+async function main() {
+    const session = await sdk.login('myapp') // login also restores existing session unlike anchor-link, can have option for forcing re-login
+
+    // Loads account
+    const account = await session.accounts[0].get()
+    // Or you can load the account from the account name with either
+    // const account = await session.accounts.get('teamgreymass')
+    // or
+    // const account = await Account.get('eos', 'teamgreymass')
+
+    // Loads account balance
+    const balance = await account.getBalance('EOS')
+
+    // Load account resources
+    const { cpu, net, ram } = await account.getResources()
```

Comment:
I think it'd already have these loaded, no need for the `await`. It'd probably also be:

```js
const { cpu, net, ram } = account.resources
```

The account object from v1/chain/get_account will have that data already.

Reply:
hmm I was thinking that the initial Account initialization wouldn't actually fetch anything and then subsequent method calls (like this one) could do the fetching. We could have some kind of caching involved so the same API request doesn't need to happen twice.

Reply:
Two ways to go, either we assume Account objects are loaded when instantiated and offer a Account.load and Account.from(loadedData), or the Account object is just representing a network + account name and you call methods to load data.

I'm leaning towards the latter option since users would have to deal with stale data, refreshing it etc. Another drawback of requiring data at instantiation is that we need to make all network calls up front and there could be more than one to get the data we need

Reply:
I took a stab at rewriting this with latter option, where the `Account` itself is just a representation of a `Name` and `ChainType` (or something similar). 

There's now a `loadData` method on the object that'll fill it with data. I'm not sure if we want to go that route or have it return some other type of object.

Reply:
Closing this thread and starting a new thread on the new piece of code.